src/utils/index.js

```diff
@@ -15,7 +15,9 @@ export function getDataTransferItems(event) {
     } else if (dt.items && dt.items.length) {
       // During the drag even the dataTransfer.files is null
       // but Chrome implements some drag store, which is accesible via dataTransfer.items
-      dataTransferItemsList = dt.items
+      // Allow only files and retun the items as a list of File,
+      // see https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem for details
+      dataTransferItemsList = Array.prototype.filter.call(dt.items, file => file.kind === 'file')
```

Comment:
So this change made me wondering if `hasFiles` is going to break folder drop integration.

Reply:
Folder drop is not something that is in the official [spec](https://www.w3.org/TR/2011/WD-html5-20110525/dnd.html) as far as I know.

So, in my opinion, supporting this could be a risk if it will never be supported natively.

Reply:
Folder drop is supported over plugins. See https://github.com/react-dropzone/react-dropzone/pull/616

Reply:
To stay on the safe side, I'd propose to leave this part of code as is and do the check in the `hasFiles` since it's the only place where it's really relevant. Ideally, if you could help building an example with folders support, so we can test against it in the browser, would be even better. Let me know if you up to it and/or need any help with that. Thoughts?

Reply:
Fair enough. I'll try to make these changes and add an example within the next few days.

Reply:
The documentation PR #662 has been merged.

Reply:
Great. I moved the filtering logic in the `hasFiles` fn, but I also made sure that the default `getDataTransferItems` only returns `File` objects (by using [getAsFile](https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/getAsFile) on the item).

Should also work with folders as the fn in the https://github.com/react-dropzone/react-dropzone/pull/662 example will only return `File` objects which we expect to get.